--- plugins/carbon/carbon.c	2014-02-09 06:34:51.000000000 +0000
+++ plugins/carbon/carbon.c	2014-02-14 23:06:00.000000000 +0000
@@ -385,6 +385,12 @@
 			}
 		}
 
+		wok = carbon_write(fd, "%s%s.%s.listen_queue %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) uwsgi.shared->backlog, (unsigned long long) now);
+		if (!wok) goto clear;
+
+		wok = carbon_write(fd, "%s%s.%s.listen_queue_errors %llu %llu\n", u_carbon.root_node, u_carbon.hostname, u_carbon.id, (unsigned long long) uwsgi.shared->backlog_errors, (unsigned long long) now);
+		if (!wok) goto clear;
+
 		usl->healthy = 1;
 		usl->errors = 0;
 
