--- Tests/test_imagefont.py	2014-05-06 17:43:49.131471706 +0000
+++ Tests/test_imagefont.py	2014-05-06 18:10:10.635052292 +0000
@@ -87,7 +87,7 @@
 	# some versions of freetype have different horizontal spacing.
 	# setting a tight epsilon, I'm showing the original test failure
 	# at epsilon = ~38.
-    assert_image_similar(im, target_img,.5)
+    assert_image_similar(im, target_img, 38)
 
 
 def test_rotated_transposed_font():
