diff -up Python-2.7.2/Modules/Setup.dist.notk Python-2.7.2/Modules/Setup.dist
--- Python-2.7.2/Modules/Setup.dist.notk	2012-02-17 15:17:57.000000000 -0800
+++ Python-2.7.2/Modules/Setup.dist	2012-02-17 15:20:33.000000000 -0800
@@ -322,7 +322,7 @@ timing timingmodule.c
 # every system.
 
 # *** Always uncomment this (leave the leading underscore in!):
-_tkinter _tkinter.c tkappinit.c -DWITH_APPINIT \
+#_tkinter _tkinter.c tkappinit.c -DWITH_APPINIT \
 # *** Uncomment and edit to reflect where your Tcl/Tk libraries are:
 #	-L/usr/local/lib \
 # *** Uncomment and edit to reflect where your Tcl/Tk headers are:
@@ -332,7 +332,7 @@ _tkinter _tkinter.c tkappinit.c -DWITH_A
 # *** Or uncomment this for Solaris:
 #	-I/usr/openwin/include \
 # *** Uncomment and edit for Tix extension only:
-	-DWITH_TIX -ltix \
+#	-DWITH_TIX -ltix \
 # *** Uncomment and edit for BLT extension only:
 #	-DWITH_BLT -I/usr/local/blt/blt8.0-unoff/include -lBLT8.0 \
 # *** Uncomment and edit for PIL (TkImaging) extension only:
@@ -341,7 +341,7 @@ _tkinter _tkinter.c tkappinit.c -DWITH_A
 # *** Uncomment and edit for TOGL extension only:
 #	-DWITH_TOGL togl.c \
 # *** Uncomment and edit to reflect your Tcl/Tk versions:
-	-ltk -ltcl \
+#	-ltk -ltcl \
 # *** Uncomment and edit to reflect where your X11 libraries are:
 #	-L/usr/X11R6/lib \
 # *** Or uncomment this for Solaris:
@@ -351,7 +351,7 @@ _tkinter _tkinter.c tkappinit.c -DWITH_A
 # *** Uncomment for AIX:
 #	-lld \
 # *** Always uncomment this; X11 libraries to link with:
-	-lX11
+#	-lX11
 
 # Lance Ellinghaus's syslog module
 #syslog syslogmodule.c		# syslog daemon interface
