--- src/collectors/postgres/postgres.py.orig	2014-05-11 19:47:22.000000000 +0100
+++ src/collectors/postgres/postgres.py	2014-09-22 21:55:50.443699598 +0100
@@ -460,6 +460,14 @@
     """
 
 
+class DatabaseReplicationStats(QueryStats):
+    path = "database.replication.%(metric)s"
+    multi_db = False
+    query = """
+        SELECT EXTRACT(epoch FROM current_timestamp - pg_last_xact_replay_timestamp()) as replay_lag
+    """
+
+
 metrics_registry = {
     'DatabaseStats': DatabaseStats,
     'DatabaseConnectionCount': DatabaseConnectionCount,
@@ -478,16 +486,19 @@
     'UserConnectionCount': UserConnectionCount,
     'TableScanStats': TableScanStats,
     'TupleAccessStats': TupleAccessStats,
+    'DatabaseReplicationStats': DatabaseReplicationStats,
 }
 
 registry = {
     'basic': (
         'DatabaseStats',
         'DatabaseConnectionCount',
+        'DatabaseReplicationStats',
     ),
     'extended': (
         'DatabaseStats',
         'DatabaseConnectionCount',
+        'DatabaseReplicationStats',
         'UserTableStats',
         'UserIndexStats',
         'UserTableIOStats',
