--- src/collectors/postgres/postgres.py	2014-01-03 20:44:43.000000000 +0000
+++ src/collectors/postgres/postgres.py	2014-02-14 23:20:23.000000000 +0000
@@ -450,6 +450,17 @@
     """
 
 
+class DatabaseReplicationStats(QueryStats):
+    path = "database.replication.%(metric)s"
+
+    def fetch(self):
+        slave_conn = self.connections.values()[0]
+        cursor = slave_conn.cursor()
+        cursor.execute("select extract('epoch' from now()) - extract('epoch' from pg_last_xact_replay_timestamp())")
+        replay_lag = cursor.fetchone()[0]
+        self.data = [dict(metric='replay_lag', value=replay_lag)]
+
+
 metrics_registry = {
     'DatabaseStats': DatabaseStats,
     'DatabaseConnectionCount': DatabaseConnectionCount,
@@ -468,16 +479,19 @@
     'UserConnectionCount': UserConnectionCount,
     'TableScanStats': TableScanStats,
     'TupleAccessStats': TupleAccessStats,
+    'DatabaseReplicationStats': DatabaseReplicationStats,
 }
 
 registry = {
     'basic': (
         'DatabaseStats',
         'DatabaseConnectionCount',
+        'DatabaseReplicationStats',
     ),
     'extended': (
         'DatabaseStats',
         'DatabaseConnectionCount',
+        'DatabaseReplicationStats',
         'UserTableStats',
         'UserIndexStats',
         'UserTableIOStats',
