See https://github.com/mrjoes/flask-admin/issues/147

diff --git a/flask_admin/contrib/sqlamodel/fields.py b/flask_admin/contrib/sqlamodel/fields.py
index d758814..c5f4282 100644
--- a/flask_admin/contrib/sqlamodel/fields.py
+++ b/flask_admin/contrib/sqlamodel/fields.py
@@ -234,9 +234,6 @@ def populate_obj(self, obj, name):
 
             field.populate_obj(model, None)
 
-            # Force relation
-            setattr(model, self.prop, obj)
-
 
 def get_pk_from_identity(obj):
     # TODO: Remove me
diff --git a/flask_admin/contrib/sqlamodel/form.py b/flask_admin/contrib/sqlamodel/form.py
index a38b30a..701913b 100644
--- a/flask_admin/contrib/sqlamodel/form.py
+++ b/flask_admin/contrib/sqlamodel/form.py
@@ -467,7 +467,7 @@ def contribute(self, converter, model, form_class, inline_model):
                 InlineModelFormList(child_form,
                                     self.session,
                                     info.model,
-                                    forward_prop.key,
+                                    reverse_prop.key,
                                     **kwargs))
 
         return form_class
