https://github.com/miracle2k/flask-assets/pull/61

--- src/flask_assets.py	2013-05-29 10:24:33.000000000 +0000
+++ src/flask_assets.py	2014-02-13 16:58:57.807091516 +0000
@@ -298,10 +298,6 @@
     import argparse
     from webassets.script import GenericArgparseImplementation, CommandError
 
-    class CatchAllParser(object):
-        def parse_known_args(self, app_args):
-            return argparse.Namespace(), app_args
-
     class FlaskArgparseInterface(GenericArgparseImplementation):
         """Subclass the CLI implementation to add a --parse-templates option."""
 
@@ -358,9 +354,6 @@
             self.implementation = impl
             self.log = log
 
-        def create_parser(self, prog):
-            return CatchAllParser()
-
         def run(self, args):
             """Runs the management script.
             If ``self.env`` is not defined, it will import it from
